telescope/ERQTelescopePID.cxx

```diff
@@ -92,53 +92,39 @@ InitStatus ERQTelescopePID::Init() {
 //--------------------------------------------------------------------------------------------------
 void ERQTelescopePID::Exec(Option_t* opt) { 
   Reset();
-  
-  for (const auto itTrackBranches : fQTelescopeTrack) {
-
-    LOG(DEBUG) << " Work with traks in " << itTrackBranches.first << FairLogger::endl;
-
-    for (Int_t iTrack(0); iTrack < itTrackBranches.second->GetEntriesFast(); iTrack++){
-
-      ERQTelescopeTrack* track = (ERQTelescopeTrack*)itTrackBranches.second->At(iTrack);
-
-      for (const auto itParticesBranches : fQTelescopeParticle[itTrackBranches.first]){
-
-        Int_t pdg = itParticesBranches.first;
-
-        //sum edep on road from telescope hit produce station to CsI and edep in CsI digi
-        Double_t csIedep = FindCsIEdepByTrack(track,pdg);
-        
-        Float_t T = track->GetSumEdep() + csIedep;
-        Double_t deadEloss = CalcEloss(itTrackBranches.first,track, pdg,T);
-
+  for (const auto& itTrackBranches : fQTelescopeTrack) {
+    const auto& station = itTrackBranches.first;
+    const auto& stationTracks = itTrackBranches.second;
+    LOG(DEBUG) << " Work with traks in " << station << FairLogger::endl;
+    for (Int_t iTrack(0); iTrack < stationTracks->GetEntriesFast(); iTrack++){
+      const auto* track = (ERQTelescopeTrack*)stationTracks->At(iTrack);
+      for (const auto itParticesBranches : fQTelescopeParticle[station]){
+        const Int_t pdg = itParticesBranches.first;
+        TVector3 backPropagationStartPoint = FindBackPropagationStartPoint(track);
+        // Calc eloss on path form start point to target
+        // In case of sensetive volume - get eloss from digi
+        // otherwise - use G4EmCalculator
+        Double_t T = CalcEloss(station, track, backPropagationStartPoint, pdg);
         //mass 
         G4IonTable* ionTable = G4IonTable::GetIonTable();
         G4ParticleDefinition* ion =  ionTable->GetIon(pdg);
         Float_t mass = ion->GetPDGMass()/1000.; //GeV
-
         //LotentzVector on telescope
         Double_t P = sqrt(pow(T,2) + 2*mass*T);
         TVector3 direction = track->GetTelescopeVertex()-track->GetTargetVertex();
         TLorentzVector lvTelescope (P*sin(direction.Theta())*cos(direction.Phi()),
                                     P*sin(direction.Theta())*sin(direction.Phi()),
                                     P*cos(direction.Theta()),
                                     sqrt(pow(P,2)+pow(mass,2)));
-        //LorentVector on target
-        T += deadEloss;
-        P = sqrt(pow(fT,2) + 2*mass*fT);
+        P = sqrt(pow(T,2) + 2*mass*T);
```

Comment:
I think we don't need to calculate 'P' twice (on the 113th line for the first time)